gc.c

```diff
@@ -13264,7 +13264,7 @@ gc_profile_record_get(VALUE _)
         gc_profile_record *record = &objspace->profile.records[i];
 
         prof = rb_hash_new();
-        rb_hash_aset(prof, ID2SYM(rb_intern("GC_FLAGS")), gc_info_decode(0, rb_hash_new(), record->flags));
+        rb_hash_aset(prof, ID2SYM(rb_intern("GC_FLAGS")), gc_info_decode(objspace, rb_hash_new(), record->flags));
```

Comment:
I don't know if passing `objspace` is correct fix or not.
I'm also wondering which change since 3.2.2 is causing this SEGV.